Extension's User Interface

FlyingKoala01 opened the pull request:
hehe

royalmo commented on html/landpage.html:
```diff
@@ -0,0 +1,10 @@
+  <body>
+    <h1>Hello World!</h1>
+    
+    <p>Isaac bobo</p>
```
User Interface missing ._.

royalmo reviewed (changes requested):
Work still needs to be done!

royalmo commented on html/landpage.html:
```diff
@@ -0,0 +1,123 @@
+  <head>
+  </head>
+  <body>
+                                        <!-- LANDPAGE -->
```
Maybe these comments could be at the beggining of the line (you can miss them right now).

royalmo reviewed (changes requested):
You should also add some screenshots @FlyingKoala01 .

This way, we can see the progress without cloning your branch every time, and it's more easy to give you some feedback.

royalmo commented on html/landpage.html:
```diff
@@ -0,0 +1,127 @@
+      <div id="contributeDiv">
+        <h2 id="aboutSubTitle">What to contribute?</h2>
+        <p>
+          LoYou can report bugs, suggestions, or
```
LoYou ? xD

royalmo commented:
@FlyingKoala01 I'm still waiting for some screenshots 😢

royalmo reviewed (changes requested):
- `html/landpage.css` should be in `css/landpage.css`

- You should maybe change the name for `user_inteface.html`, `user_interface.css`, and `user_interface.js`.

- Still waiting for some screenshots 😢

FlyingKoala01 commented:
![about](https://user-images.githubusercontent.com/98028789/177590224-95ae04a3-402e-4376-91aa-95516d0883be.PNG)
![landpage](https://user-images.githubusercontent.com/98028789/177590227-2db597f7-31ca-4d6e-924e-259fe4451325.PNG)
![settings_off](https://user-images.githubusercontent.com/98028789/177590230-fa389bba-74cd-45b7-96a8-f46ee64a2d82.PNG)
![settings_on](https://user-images.githubusercontent.com/98028789/177590232-dea7062b-8572-4e70-98f7-5817002e423c.PNG)
Here are a few screenshots as required.
As you can see, there is still a lot to get done. Indentation, color-choice, font-size for titles, paragraphs etc. Also we have to talk a bit more in detail about the specific layout for the landpage.

royalmo commented on css/user_interface.css:
```diff
@@ -0,0 +1,63 @@
+    text-align: justify;
+    margin-right: 10%;
+    margin-left: 10%;
+    text-decoration: underline;
```
I think i wouldn't underline the subtitles.

royalmo reviewed (changes requested):
In a summary:
- WTF have you done in manifest.json xD
- Change file name from UI.js to user_interface.js
- Change _credits_ title, put picture, made by centered.
- Put just the back button, on top, in the about page.

royalmo commented:
@FlyingKoala01 I have reviewed the code. This way, now you have like a TO-DO list. Mark each conversation as resolved when you finish fixing each thing.

FlyingKoala01 commented on html/user_interface.html:
```diff
@@ -0,0 +1,121 @@
+      <h1 class="title">About</h1>
+    </div>
+    <div class="bodyPopup">
+      <h2 id="aboutSubTitle">Credits</h2>
```
Where exactly do you want the iTIC logo? In the first popup window like we agreed on the paint image from day 0, or do you want it somewhere else? 
"Here was meant to be Josep's logo, with as an image footer" is confusing me as in where you want the logo.

FlyingKoala01 commented:
Maybe we could add an introductory msg in the first popup window. Maybe with a random quote or something to fill the blank space. Otherwise, if we only add links and maybe titles(?), will be lots of useless css to not make it look like a blank window

FlyingKoala01 commented:
![about](https://user-images.githubusercontent.com/98028789/178540495-1d7fb311-331d-4d7e-9a9b-0e3fbd6749ed.PNG)
![landpage](https://user-images.githubusercontent.com/98028789/178540499-9981e660-6097-439b-ad2f-115299fc78d5.PNG)
![settings](https://user-images.githubusercontent.com/98028789/178540502-a68ae5a7-98d3-4cfd-839b-f28cd332e599.PNG)
What do you think?
I tried to have a quick fix around the logo's size. I think if we make the title space bigger while adding a gradient to not make the bigger size as obvious works as a decent temporary solution to the issue. Also it does look better, not as basic as it was before. Also I tried to make the buttons to look a bit better. Can't show you on hover effect with screenshots but it's nothing spectacular, just shading and some color-changing stuff.
Also I resized the popup. Height went from 500px to 400px. There was definitely too much space, we can always reverse this but for the moment I think it's for the best to keep it smaller and not have the idea of filling white space in the back of our minds. 
Anyways, I think we could keep it like this if you like it. <3<3

royalmo commented on manifest.json:
```diff
@@ -3,25 +3,30 @@
     "description": "__MSG_extension_description__",
     "version": "1.6",
     "manifest_version": 2,
```
Did you change the manifest.json things that I mentionned? I still see them changed ...

royalmo reviewed (changes requested):
Perfect. Just change the manifest.json things that I stated on earlier comments, and change the logo assets as I also mentionned earlier.

Just a thing, maybe you could put the logo in `assets/user_interface/logo_icon.png` and `assets/user_interface/logo_banner.png`, it's more intuitive. (yes, I know, I didn't do it with the original logos xD)

royalmo commented:
Oh, and forgot to mention: did you made some transitions between the landpage and about/settings. If yes, great!, If no, don't worry, we'll make them once the whole user-interface is finished, in another PR. I think it isn't part of the UI MVP (Minimum Valuable Product).

FlyingKoala01 commented:
![first_option](https://user-images.githubusercontent.com/98028789/178940441-79c73f5a-26d1-4c72-8d85-a808c743d85f.PNG)
![second_option](https://user-images.githubusercontent.com/98028789/178940445-e9a0b632-8e8b-4e10-9c54-83f86baf63e1.PNG)
So I tried to put the modified logo on top and I honestly don't really like it. If you think it's fine we can leave it there. I also tried to place it at the bottom, I think if we have to put the logo, it has to go to the bottom. Otherwise, we'd be filling the top section with technically useless stuff, even if it's the about section. Up to you, tho. 

![landpage](https://user-images.githubusercontent.com/98028789/178940694-706fe9cf-4e02-4a69-81e5-f234b18c42c1.PNG)
Secondly, I placed the logo and added an h1 there as we mentioned. I think it looks better with the banner itself, what do you think?


Alsooo, I added transitions (checkout user_interface.js) they look honestly good, but we there is a tiny initial bug (Who would have thought, right? 😢 ). But yea, if we fix this, it will look real smooth. 

Next, I would try and polish the landpage. How do we deal with the hyperlinks? They look ugly. I tried to look-up for some css nerd tricks to make it look more appealing but it's just bad. Do I place them in a table and just deal with it? At the end of the day, we look for functionality, right?

Once we are done with the landpage, we just gotta go through the settings section and then merge :3

royalmo reviewed (changes requested):
Nice work! In a comment below I will state the changes.

royalmo commented on css/user_interface.css:
```diff
@@ -6,6 +6,7 @@ body {
     margin: 0px;
     font-family: Verdana, sans-serif;
     background-color: #F4F4F4;
+    border: 1px solid black;
```
I placed the border automaticlly with powerpoint, but maybe without border and a shadow it would look better (but I assume that it's harder to implement).

royalmo commented on js/user_interface.js:
```diff
@@ -82,6 +82,14 @@ $(document).ready(function () {
+});
+
+
 function hello_world() {
```
Maybe you could remove this function xD

royalmo reviewed (changes requested):
- `assets/user_interface/itic-logo-no-line-resized.png`: the quality of the logo is low, maybe you could fix it. If you need a higher quality logo, i can send the originals to you.
- Maybe try another font-size/family/whatever for the text of the table.
- Things that I mentionned in the comments.

royalmo commented:
After talking in Discord, here you have a complete checklist:

- [ ] Reduce image quality to minimum required.
- [ ] Make landpage image logo round, not oval.
- [ ] OCW Logo on landpage links. You could ask Josep for that.
- [ ] Colors on landpage links.
- [ ] Animation on hover in landpage links.
- [ ] Fix bug when switching from tabs.
- [ ] Fade out of landpage instead of blackout.
- [ ] Settings page: finish it & store data in chrome storage.

FlyingKoala01 commented:
![landpage](https://user-images.githubusercontent.com/98028789/180416170-78f71027-4a3d-48f4-b619-f34c92366e2d.PNG)
![settings](https://user-images.githubusercontent.com/98028789/180416173-0160cdd2-18ba-4f7c-8013-52fb82429d88.PNG)
![settings1](https://user-images.githubusercontent.com/98028789/180416177-4fd3bdda-7e88-459c-ae9a-bb36777507fc.PNG)
![settings2](https://user-images.githubusercontent.com/98028789/180416179-7cf34231-0bf0-4c62-8f21-de1391dba08d.PNG)
How does it look so far? :D

royalmo reviewed (changes requested):
Nice!!

Some details:
- Try these switches, I think they look less plastic: https://codepen.io/alvarotrigo/pen/NWweaqW
- Does the scrool bar have a 1px margin? O.o
- The NEW needs to be a little bit shiny or something :)
- The _save UPC credentials must be enabled_ maybe in gray, to make it less appealing

Thx <3<3

royalmo commented on css/user_interface.css:
```diff
@@ -0,0 +1,251 @@
+    ::-webkit-scrollbar-thumb {
+    border-radius: 10px;
+	background-color: #0d6efd;	
+}
```
This part isn't correctly indented.

royalmo commented on css/user_interface.css:
```diff
@@ -0,0 +1,251 @@
+}
+.tableLinksTr {
+    height: 50px;
+    cursor: pointer;
```
Please don't do like me xD Put the same separation between each rule :)

royalmo commented on css/user_interface.css:
```diff
@@ -0,0 +1,251 @@
+    position: absolute;
+    content: "";
+    height: 20px;
+    width: 20px;
```
Is it me or all of this has a two space indent?

royalmo commented on js/user_interface.js:
```diff
@@ -0,0 +1,145 @@
+        for (const value of checkboxes.values()){
+            checkboxesId.push(value.id);
+            window.localStorage.setItem(value.id, String(document.getElementById(value.id).checked));
+        }
```
function storeCheckboxes could (and should) go outside $(document).ready().

royalmo reviewed (changes requested):
Good job!
Just layed some comments, but all ok.

FlyingKoala01 commented:
:3

royalmo reviewed (changes requested):
Almost done!

Just resize for me the Gmail and OCW logos to a fixed size :)

royalmo reviewed (approved):
🥳
